### DISCUSSION TITLE
How to Scroll in Treeview?

### DISCUSSION THREAD

**Discussion by jdgarciaa11** [enisn/UraniumUI#293] [Q&A]
I have a problem with scroll in a tree view implementation, its a problem with de tree view or my implementation
<VerticalStackLayout>
                <ScrollView>
                    <Frame>
                        <material:TreeView 
                         ItemsSource="{Binding Nodes}"                  
                         LoadChildrenCommand="{Binding LoadChildrenCommand}">
                            <material:TreeView.ItemTemplate>
                                <DataTemplate>
                                    <HorizontalStackLayout Spacing="5" VerticalOptions="Center">
                                        <Image>
                                            <Image.Triggers>
                                                <DataTrigger TargetType="Image" Binding="{Binding IsDirectory}" Value="True">
                                                    <Setter Property="Source" Value="{FontImageSource FontFamily=MaterialRegular, Glyph={x:Static m:MaterialRegular.Folder}, Color={AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource PrimaryDark}}}" />
                                                </DataTrigger>
                                                <DataTrigger TargetType="Image" Binding="{Binding IsDirectory}" Value="False">
                                                </DataTrigger>
                                            </Image.Triggers>
                                        </Image>
                                        <Label Text="{Binding Name}" FontAttributes="Bold" >
                                            <Label.GestureRecognizers>
                                                <DragGestureRecognizer DragStartingCommand="{Binding Source={x:RelativeSource AncestorType={x:Type model:treeVM}},Path=CogerInformacion}" CanDrag="{Binding Source={x:RelativeSource AncestorType={x:Type model:treeVM}},Path=CanDrag}" DragStartingCommandParameter="{Binding }"></DragGestureRecognizer>
                                            
                                            </Label.GestureRecognizers>
                                            
                                        </Label>
                                    </HorizontalStackLayout>
                                </DataTemplate>
                            </material:TreeView.ItemTemplate>
                        </material:TreeView>
                    </Frame>
                </ScrollView>
            </VerticalStackLayout>

**Comment by enisn** [comment]
You can put the TreeView inside a ScrollView:

```xml
<ScrollView>
<material:TreeView  />
</ScrollView>
```

Scrolling works as expected like in this page: https://github.com/enisn/UraniumUI/blob/develop/demo/UraniumApp/Pages/TreeViews/TreeViewFileSystemPage.xaml
